src/parsers/srdf.hxx

```diff
@@ -242,30 +242,37 @@ namespace pinocchio
           if( name == "half_sitting")
           {
             // Iterate over all the joint tags
-            BOOST_FOREACH(const ptree::value_type & joint, v.second)
+            BOOST_FOREACH(const ptree::value_type & joint_tag, v.second)
             {
-              if (joint.first == "joint")
+              if (joint_tag.first == "joint")
               {
-                std::string joint_name = joint.second.get<std::string>("<xmlattr>.name");
-                const Scalar joint_config = (Scalar)joint.second.get<double>("<xmlattr>.value");
-                if (verbose)
-                {
-                  std::cout << "(" << joint_name << " , " << joint_config << ")" << std::endl;
-                }
-                // Search in model the joint and its config id
+                std::string joint_name = joint_tag.second.get<std::string>("<xmlattr>.name");
                 typename Model::JointIndex joint_id = model.getJointId(joint_name);
-
+                
+                // Search in model the joint and its config id
                 if (joint_id != model.joints.size()) // != model.njoints
                 {
                   const JointModel & joint = model.joints[joint_id];
-                  assert(joint.nq() == 1 && "SRDF:half_sitting only handles 1 DoF joints");
-                  model.neutralConfiguration(joint.idx_q()) = joint_config; // joint with 1 dof
-                  // model.neutralConfiguration.segment(joint.idx_q(),joint.nq()) = joint_config; // joint with more than 1 dof
+                  typename Model::ConfigVectorType joint_config(joint.nq());
+                  if(joint.nq() == 1)  joint_config[0] = (Scalar)joint_tag.second.get<double>("<xmlattr>.value");
+                  else
+                  {
+                    const std::string joint_val = joint_tag.second.get<std::string>("<xmlattr>.value");
+                    std::istringstream config_string(joint_val);
+                    const std::vector<double> config_vec((std::istream_iterator<double>(config_string)), std::istream_iterator<double>());
+                    joint_config = ModelTpl<Scalar,Options,JointCollectionTpl>::ConfigVectorType::Map(config_vec.data(), config_vec.size());
+                  }
+                  model.neutralConfiguration.segment(joint.idx_q(),joint.nq()) = joint_config; // joint with more than 1 dof
```

Comment:
You can also remove the ending comment which is now outdated!

Reply:
Done with force push.